### ISSUE TITLE
Tests failing with python 3

### ISSUE THREAD

**Issue by rdoursenaud** [Gandi/gandi.cli#181]
All tests pass with python 2 but when running them with python 3 (3.5.1) I get the following results:
```
======================================================================
FAIL: test_create_csr_empty (gandi.cli.tests.commands.test_certificate.CertTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "gandi.cli/gandi/cli/tests/commands/test_certificate.py", line 421, in test_create_csr_empty
    self.assertEqual(result.output, '')
AssertionError: "gandi[163 chars]()\n" != ''
- gandi.cli/gandi/cli/modules/cert.py:384: ResourceWarning: unclosed file <_io.TextIOWrapper name='sandbox/example.txt' mode='r' encoding='UTF-8'>
-   csr = open(csr).read()


----------------------------------------------------------------------
```
This prevents the updated Arch Linux package I was preparing from installing.

**Comment by sayoun** [comment]
I updated the tox.ini file for testing in 3.5 but did not update the Travis file, which is done now, I'll see if I can reproduce it.

**Comment by sayoun** [comment]
Ok Travis run is finished and no error were reported for 3.5 tests https://travis-ci.org/Gandi/gandi.cli/jobs/120577395 so I don't know..

**Comment by rdoursenaud** [comment]
Hi @sayoun. Thanks for having a look at this.

I notice that in Travis you use ```python setup.py nosetests``` wheras in my package I was using ```python setup.py test```.

Of course the warning is not issued with nosetests. I switched to nosetests in the package for the moment but I wonder if the warning wasn't legitimate since this file is opened and returned but never closed.

**Comment by sayoun** [comment]
@rdoursenaud hello, good that you found a way to fix your package, but you are right, I should fix the code to properly close this file anyway.

**Comment by sayoun** [comment]
Will be fixed in next release.

**Comment by sayoun** [comment]
Fixed in release 0.19
